.gitignore

```diff
@@ -0,0 +1,115 @@
+# Created by https://www.gitignore.io/api/python
+# Edit at https://www.gitignore.io/?templates=python
+
+### Python ###
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
+
+# C extensions
+*.so
+
+# Distribution / packaging
+.Python
+build/
+develop-eggs/
+dist/
+downloads/
+eggs/
+.eggs/
+lib/
+lib64/
+parts/
+sdist/
+var/
+wheels/
+pip-wheel-metadata/
+share/python-wheels/
+*.egg-info/
+.installed.cfg
+*.egg
+MANIFEST
+
+# PyInstaller
```

Comment:
Do we need PyInstaller? This is a library so I don't think anyone will ever compile it to a binary. Did you have smt else in mind?

Reply:
This comes from http://gitignore.io/api/python, which covers a wide range of common Python files that never should be added in, based on common community practices. As mentioned in a comment below, I think relying on these opinionated community-built tools to adhere to common community practices could be better, as then we never have to think about them in the future. But if you still disagree, let me know, I'll remove the ones you request to remove.

Reply:
While I do agree on common practices, I prefer not to add lines that will not be relevant to the project in a foreseeable future. I usually start with a core of known offenders in the context of the library and build from there. I remember once seeing `.tox` in a project's gitignore and spending 5 minutes to make sure that I did not miss anything; tox was indeed not used in the project, the gitignore was just copy-pasted :)

I saw you edited [PyMC4's gitignore](https://github.com/pymc-devs/pymc4/blob/master/.gitignore), it is very well organized. Could we add the ones related to IDEs here? And maybe also `.env` and `.venv`?

Reply:
Just added them, will be pushing shortly.